sml-bin/redeploy-testlib.sml

```diff
@@ -0,0 +1,45 @@
+open OS.Path;
+open OS.FileSys;
+
+fun printLn x = print (x ^ "\n")
+
+fun fileExists x = access (x,[])
+
+fun testLibName base slug =
+    joinDirFile
+        {
+          dir = (concat (base,slug))
+        , file = "testlib.sml"
+        }
+
+fun dirList dirname = let
```

Comment:
Can you make the formatting more consistent with the other sml-files (e.g. testlib.sml)? E.g. always indent by two spaces and for example
```
fun foo = 
  let
    val a = ...
  in
    do_something ...
  end; (* Note: there is a semicolon. *)
```

instead of
```
fun foo = let
    val a = ...
in
    do_something ...
end;
```
I have the feeling that this is better. But feel free to criticize this :) .

Reply:
Sure.  I'll make some changes.  I need to figure out how to set sml-mode to use 2 spaces.  It's indentation is worse than indentation in most emacs major modes.  There really shouldn't be semicolons at the end of function definitions like this.  Semicolons are for sequencing typically imperative actions.  I had them in the first version of this script, but I think we should avoid them except where they are meaningful.

Reply:
Personally, I don't think there is an issue with let being on the opening line of the function, but I don't mind changing it for consistency.

---

I write a lot of haskell at work, so I am addicted to commas to the left in records and lists ;p

Reply:
OK I am fine with omitting the semicolons. I was used to them since in the Book "ML for the working programmer" the author uses them wherever they are allowed (not just in the REPL or for imperative actions). But he actually states somewhere that they are optional there. It seems to be a matter of taste. I followed the author in my code, since I was used to something similar in javascript: In js "most of the time" the semicolons are not relevant, but not always, and therefore the general advice for js is to put them everywhere (I know javascript is probably not the most well designed language :D).

So since testlib already omits them we should omit them everywhere.

Reply:
Yeah, with JS including the semicolons makes sure everything works as intended.  With SML, including semicolons can change the meaning of a program :exploding_head: